share/smack/lib/math.c

```diff
@@ -125,49 +126,50 @@ float copysignf(float x, float y) {
 
 int __isnormalf(float x) {
   int ret = __VERIFIER_nondet_int();
-  __SMACK_code("@ := if $isnormal.bvfloat(dtf(@)) then 1bv32 else 0bv32;", ret, x);
+  __SMACK_code("@ := if $isnormal.bvfloat.bool(@) then $1 else $0;", ret, x);
   return ret;
 }
 
 int __isSubnormalf(float x) {
   int ret = __VERIFIER_nondet_int();
-  __SMACK_code("@ := if $issubnormal.bvfloat(dtf(@)) then 1bv32 else 0bv32;", ret, x);
+  __SMACK_code("@ := if $issubnormal.bvfloat.bool(@) then $1 else $0;", ret, x);
   return ret;
 }
+*/
 
 int __iszerof(float x) {
   int ret = __VERIFIER_nondet_int();
-  __SMACK_code("@ := if $iszero.bvfloat(dtf(@)) then 1bv32 else 0bv32;", ret, x);
+  __SMACK_code("@ := if $iszero.bvfloat.bool(dtf(@)) then $1 else $0;", ret, x);
```

Comment:
I don’t understand what you mean by:

> types of variadic arguments to variadic functions would be prompted. For this case, x would be lifted to a double.

Also, what about the lack of `dtf` at previous uses, on Lines 40, 129, and 135. Are these lines incorrect as they are? Why has the call to `dtf` been removed there?

Reply:
Sorry for the confusion. There shouldn't be any inconsistency in using `dtf`. Line 40, 129, 135 are not correct. Calls to `dtf` in these locations are mistakenly removed as a previous misunderstanding of why `dtf` is used here.

We should add `dtf` calls to make Boogie type check because arguments of variable argument function may be changed to more precise types. https://stackoverflow.com/questions/11270588/variadic-function-va-arg-doesnt-work-with-float

Reply:
So we are going to fix this before approval right?

Reply:
Aren't these lines commented out?
Basically, @shaobo-he noticed other problems with a bunch of functions in `math.c`, and he commented those out in this pull request. @liammachado has been working on a proper fix for all the commented out functions.

Reply:
Oh, that is true; the PR’s code diff did not make that clear. I retract this point then.